src/prefect/client/orion.py

```diff
@@ -1317,8 +1315,28 @@ async def create_deployment(
             parameter_openapi_schema=parameter_openapi_schema,
         )
 
+        if work_pool_name is not None:
+            deployment_create.work_pool_name = work_pool_name
+        if work_pool_queue_name is not None:
+            deployment_create.work_pool_queue_name = work_pool_queue_name
+
+        json = deployment_create.dict(json_compatible=True, exclude_unset=True)
+        # exclude_unset causes the anchor date to be regenerated. Producing a dict
+        # for the schedule individually prevents the anchor date from being regenerated
+        # when the schedule is a pydantic model.
+        if isinstance(
+            schedule,
+            (
+                schemas.schedules.CronSchedule,
+                schemas.schedules.IntervalSchedule,
+                schemas.schedules.RRuleSchedule,
+            ),
+        ):
+            json["schedule"] = schedule.dict(json_compatible=True)
```

Comment:
This makes me uncomfortable because we've now bypassed any validator mutation of the schedule on deployment create. It could be a very sneaky bug in the future. Can we change the above `exclude_unset` to `exclude={...}` and target the specific problem fields by checking if they are in `__fields_set__`?

Reply:
Make sense. I believe I've implemented that approach. If it looks good to you, let me know and I'll merge this PR.